mongodb/src/main/java/org/apache/metamodel/mongodb/MongoDbDataContext.java

```diff
@@ -435,6 +441,19 @@ private String getOperatorName(FilterItem item) {
         return operatorName;
     }
 
+    private Pattern turnOperandIntoRegExp(Object operand) {
+        String operandAsString = operand.toString();
+        StringBuilder operandAsRegExp = new StringBuilder(replaceWildCardLikeChars(operandAsString));
+        if (!operandAsString.startsWith("%") && operandAsString.contains("%")) {
+            operandAsRegExp = operandAsRegExp.insert(0, "^");
+        }
+        return Pattern.compile(operandAsRegExp.toString(), Pattern.CASE_INSENSITIVE);
+    }
+
+    private String replaceWildCardLikeChars(String operand) {
+        return operand.replace("%","");
```

Comment:
Shouldn't this replace with something like ```.*```? Or else it seems to me it is just removing the % symbol?